mono/utils/mono-hwcap-ppc.c

```diff
@@ -62,16 +62,27 @@ mono_hwcap_arch_init (void)
 			mono_hwcap_ppc_has_multiple_ls_units = TRUE;
 	}
 #elif defined(_AIX)
-	/*
-	 * FIXME: ensure these are valid, and we match Linux ones
-	 */
-	mono_hwcap_ppc_is_isa_2x = __power_4_andup() ? TRUE : FALSE;
-	mono_hwcap_ppc_is_isa_64 = __cpu64() ? TRUE: FALSE;
+	if (__cpu64())
+		mono_hwcap_ppc_is_isa_64 = TRUE;
```

Comment:
Is 64bit ness really a runtime check?

Reply:
AIX runs 32-bit and 64-bit userland under the same kernel, always. We're compiled 64-bit, but nothing really stops us from being 32-bit for whatever reason.

Reply:
Can't we tell 32bit from 64bit with ifdef or from sizeof(void*)? Or can 32bit code take advantage of being on a 64bit system?

Reply:
This seems to be consistent with how Linux does it (check for 64-bit feature flag) - perhaps that is the case? I would assume say the JIT would know to use the native pointer width instead of relying on a flag that may not even be set properly (seeing as this file doesn't have a path for say, Mac OS or NetBSD)

Reply:
It appears to be an optimization for ppc32 in integer to float conversions. If the processor is actually 64bit, then 32bit assumes the fcfid instruction is available.

https://www.ibm.com/support/knowledgecenter/ssw_aix_72/com.ibm.aix.alangref/idalangref_fcfid_instrs.htm